UPGRADE.md

```diff
@@ -37,6 +37,158 @@ are terminated. In this way, Kong will serve new requests via the new
 configuration, without dropping existing in-flight connections.
 
 
+## Upgrade to `2.1.0`
+
+Kong adheres to [semantic versioning](https://semver.org/), which makes a
+distinction between "major", "minor" and "patch" versions. The upgrade path
+will be different on which previous version from which you are migrating.
+
+If you are migrating from 2.0.0, upgrading into 2.1.x is a minor upgrade,
+but read below for important instructions on database migration, especially
+for Cassandra users.
+
+If you are migrating from 1.x, upgrading into 2.1.x is a major upgrade,
+so, in addition, be aware of any [breaking changes](#breaking-changes-2.0.0)
+between the 1.x and 2.x series below, further detailed in the
+[CHANGELOG.md](https://github.com/Kong/kong/blob/2.0.0/CHANGELOG.md) document.
+
+#### 1. Dependencies
+
+If you are using the provided binary packages, all necessary dependencies
+for the gateway are bundled and you can skip this section.
+
+If you are building your dependencies by hand, there are changes since the
+previous release, so you will need to rebuild them with the latest patches.
+
+In order to use all Kong features, including the new
+dynamic upstream keepalive behavior, the required OpenResty version is
+[1.15.8.3](http://openresty.org/en/changelog-1015008.html) and the
+the set of [OpenResty
+patches](https://github.com/Kong/kong-build-tools/tree/master/openresty-build-tools/openresty-patches)
+included has changed, including the latest release of
+[lua-kong-nginx-module](https://github.com/Kong/lua-kong-nginx-module).
+Our [kong-build-tools](https://github.com/Kong/kong-build-tools)
+repository allows you to build OpenResty with the necessary patches
+and modules easily.
+
+For Go support, you also need to build both your plugins and
+the [Kong go-pluginserver](https://github.com/kong/go-pluginserver).
+The documentation includes detailed [instructions on how to build
+the plugin server and your plugins](https://docs.konghq.com/2.1.x/go/).
+
+#### 2. Template Changes
+
+The **Nginx configuration file has changed**, which means that you need to
+update it if you are using a custom template.
+
+To view the configuration changes between 2.0.0 and 2.1.0, clone the
+[Kong repository](https://github.com/kong/kong) and run `git diff`
+on the configuration templates, using `-w` for greater readability:
+
+```
+git clone https://github.com/kong/kong
+cd kong
+git diff -w 2.0.0 2.1.0 kong/templates/nginx_kong*.lua
+```
+
+To produce a patch file, use the following command:
+
+```
+git diff 2.0.0 2.1.0 kong/templates/nginx_kong*.lua > kong_config_changes.diff
+```
+
+#### 3. Suggested Upgrade Path
+
+##### Upgrade from `0.x` to `2.1.0`
+
+The lowest version that Kong 2.1.0 supports migrating from is 1.0.0.
```

Comment:
Isn't it required to upgrade from Kong 1.0 to 1.5 before upgrading to 2.x?

Reply:
Not really, as explained in the 2.0 section of this document. The 0.x migrations were squashed, but all 1.x-series migrations are there. We will squash the 1.x migrations on Kong 3.0.